cmd/kubelet/kubelet.go

```diff
@@ -45,12 +45,18 @@ func die(err error) {
 }
 
 func main() {
```

Comment:
This main is significantly different from all the other mains.

https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/proxy.go
https://github.com/kubernetes/kubernetes/blob/master/plugin/cmd/kube-scheduler/scheduler.go
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/apiserver.go
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/controller-manager.go

I particularly like the kube-proxy Options model with a signature:
```
type Options
  func NewOptions() (*Options, error)
  func (o *Options) ApplyDefaults(in *componentconfig.KubeProxyConfiguration (*componentconfig.KubeProxyConfiguration, error)
  func (o *Options) Complete() error
  func (o *Options) Run() error
  func (o *Options) Validate(args []string) error
```

In that model (*Options).New() builds the scheme and (*Options).Run() encompasses all your setup. No need to muck now but it would be nice to move to this model